Description: extract from upstream commit fix for itk4.9
From 5608ed64ee9993882e99781172568238933170d6 Mon Sep 17 00:00:00 2001
From: Hans Johnson <hans-johnson@uiowa.edu>
Date: Fri, 12 Jun 2015 21:02:28 -0500
Subject: [PATCH] BUG: ITK fix for resolution bug

This provides a working version of ANTS
for convering displacements to/from itk::Image
and itk::Transform without loss of precision.

Fixes https://github.com/stnava/ANTs/issues/74
---
 Examples/antsRegistrationOptimizerCommandIterationUpdate.h | 6 +++---
 SuperBuild/External_ITKv4.cmake                            | 3 +--
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/Examples/antsRegistrationOptimizerCommandIterationUpdate.h b/Examples/antsRegistrationOptimizerCommandIterationUpdate.h
index e34d63c..cc115ac 100644
--- a/Examples/antsRegistrationOptimizerCommandIterationUpdate.h
+++ b/Examples/antsRegistrationOptimizerCommandIterationUpdate.h
@@ -257,7 +257,7 @@ class antsRegistrationOptimizerCommandIterationUpdate : public itk::Command
         // Copy the information to each sub transform and add this transform to the final composite transform.
         const typename TransformBaseType::ParametersType & fixedImage_paras =
           inputFixedTransform->GetNthTransform(i)->GetParameters();
-        const typename TransformBaseType::ParametersType & fixedImage_fixed_paras =
+        const typename TransformBaseType::FixedParametersType & fixedImage_fixed_paras =
           inputFixedTransform->GetNthTransform(i)->GetFixedParameters();
         subTransform->SetParameters( fixedImage_paras );
         subTransform->SetFixedParameters( fixedImage_fixed_paras );
@@ -289,7 +289,7 @@ class antsRegistrationOptimizerCommandIterationUpdate : public itk::Command
         dynamic_cast<TransformBaseType *>( inputMovingTransform->GetNthTransform(i)->CreateAnother().GetPointer() ) );
       const typename TransformBaseType::ParametersType & moving_paras =
         inputMovingTransform->GetNthTransform(i)->GetParameters();
-      const typename TransformBaseType::ParametersType & moving_fixed_paras =
+      const typename TransformBaseType::FixedParametersType & moving_fixed_paras =
         inputMovingTransform->GetNthTransform(i)->GetFixedParameters();
       subTransform->SetParameters( moving_paras );
       subTransform->SetFixedParameters( moving_fixed_paras );
@@ -323,7 +323,7 @@ class antsRegistrationOptimizerCommandIterationUpdate : public itk::Command
         dynamic_cast<TransformBaseType *>( inputMovingTransform->GetNthTransform(i)->CreateAnother().GetPointer() ) );
       const typename TransformBaseType::ParametersType & moving_paras =
         inputMovingTransform->GetNthTransform(i)->GetParameters();
-      const typename TransformBaseType::ParametersType & moving_fixed_paras =
+      const typename TransformBaseType::FixedParametersType & moving_fixed_paras =
         inputMovingTransform->GetNthTransform(i)->GetFixedParameters();
       subTransform->SetParameters( moving_paras );
       subTransform->SetFixedParameters( moving_fixed_paras );
